projects/ngneat/hotkeys/src/lib/hotkeys.directive.ts

```diff
@@ -13,59 +14,45 @@ interface Options {
 }
 
 @Directive({
-  selector: '[hotkeys]'
+  standalone: true,
+  selector: '[hotkeys]',
 })
-export class HotkeysDirective implements OnChanges, OnDestroy {
+export class HotkeysDirective {
+  private destroyRef = inject(DestroyRef);
+  private hotkeysService = inject(HotkeysService);
+  private elementRef = inject(ElementRef);
   private subscription: Subscription;
 
-  constructor(private hotkeysService: HotkeysService, private elementRef: ElementRef) {}
-
-  @Input() hotkeys: string;
-  @Input() isSequence: boolean = false;
-  @Input() hotkeysGroup: string;
-  @Input() hotkeysOptions: Partial<Options> = {};
-  @Input() hotkeysDescription: string;
-
-  @Output()
-  hotkey = new EventEmitter<KeyboardEvent | Hotkey>();
-
-  ngOnChanges(changes: SimpleChanges): void {
-    this.deleteHotkeys();
-    if (!this.hotkeys) {
-      return;
-    }
-
-    const hotkey: Hotkey = {
-      keys: this.hotkeys,
-      group: this.hotkeysGroup,
-      description: this.hotkeysDescription,
-      ...this.hotkeysOptions
-    };
-
-    this.setHotkeys(hotkey);
-  }
-
-  ngOnDestroy() {
-    this.deleteHotkeys();
-  }
-
-  private deleteHotkeys() {
-    if (this.subscription) {
-      this.subscription.unsubscribe();
-    }
-    this.subscription = null;
-  }
-
-  private setHotkeys(hotkeys: Hotkey | Hotkey[]) {
-    const coercedHotkeys = coerceArray(hotkeys);
-    this.subscription = merge(
-      coercedHotkeys.map(hotkey => {
-        return this.isSequence
-          ? this.hotkeysService.addSequenceShortcut({ ...hotkey, element: this.elementRef.nativeElement })
-          : this.hotkeysService.addShortcut({ ...hotkey, element: this.elementRef.nativeElement });
-      })
-    )
-      .pipe(mergeAll())
-      .subscribe(e => this.hotkey.next(e));
+  hotkeys = input<string>();
+  isSequence = input(false);
+  hotkeysGroup = input<string>();
+  hotkeysOptions = input<Partial<Options>>({});
+  hotkeysDescription = input<string>();
+  @Output() hotkey = new EventEmitter<KeyboardEvent | Hotkey>();
+
+  constructor() {
+    effect(() => {
```

Comment:
Why do we need this change? ngOnChanges still works with signal inputs

Reply:
I am not familiar with such best practice. effect is currently in dev preview

Reply:
No. They are more stable than effect that its implementation was changed several times.

Reply:
Alright, I'll revert to ngchanges when I have time this week.


> No. They are more stable than effect that its implementation was changed several times.

@netanel-utila could you let me know how you came to this conclusion ? Is there an ”easy” way to know how confident the Angular devs are with the developer preview features ?

Reply:
experience